router/router.go

```diff
@@ -2213,31 +2212,19 @@ func (rt *HandleT) Start() {
 	rt.logger.Infof("Starting router : %s", rt.destName)
```

Comment:
shouldn't we use a lifecycle lock for controlling concurrent access to Start/Stop methods? E.g. similar to what we are doing in jobsdb?

Reply:
Currently, there is no need for concurrent support for Start/Stop. Following the standard library paradigm, I suggest not adding concurrent support unless needed. We can be explicit about concurrency support in the method documentation.

Reply:
Start performs its operations asynchronously, through a go routine, effectively returning before it is actually done. So the error we are fixing here occurred because we were calling Stop before the router was actually started, right?

Reply:
Oh know I understood the exact issue.

Start should return only once it is ready to be stopped. A loop like:

```go
for {
   router.Start()
   router.Stop()
}
```
Should work like charm.

Reply:
Yes but I don't think that should stop this PR from being merged. Its purpose is not fixing the setup/start/shutdown flow (remember that setup is also involved here). For that I had created a separate task [here](https://www.notion.so/rudderstacks/Router-setup-start-stop-80425af6cf2645b0a3d53620b8aca2f9). So I think this PR was just meant to improve the test situation with the plan to then fix the router setup/start/shutdown properly later.

Not strong on it though, just wanted to point it out. We can do it all together as well.

Reply:
IMO since the changes here are affecting non-test code, this is the time to make sure that start & stop behave correctly, as they should ;)

Reply:
Happy with that, @cisse21 what do you think?

Reply:
Yeah makes that sense too. I just raised this to unblock people from flaky tests causing them to rerun